libraries/chain/controller.cpp

```diff
@@ -2101,7 +2116,7 @@ struct controller_impl {
          trusted_producer_light_validation = old_value;
       });
       try {
-         block_state_ptr bsp = block_state_future.get();
+         EOS_ASSERT( bsp, block_validate_exception, "null block" );
```

Comment:
Will this break consensus as there was not check before? Or should `bsp` is checked before `push_block` is called?

Reply:
This is one of those cases for @greg7mdp abort. It should never happen. If b is null then core dump should happen soon if this `EOS_ASSERT` was removed.